File: html/redcap_v15.7.6/API/examples/ruby/import_project.rb
#!/usr/bin/env ruby

require 'json'
require 'curl'
require './settings.rb'
include Settings

record = {
  :project_title => 'Project ABC',
  :purpose       => 0,
  :purpose_other => '',
  :project_notes => 'Some notes about the project'
}

data = record.to_json

fields = {
  :token   => Settings::API_SUPER_TOKEN,
  :content => 'project',
  :format  => 'json',
  :data    => data,
}

ch = Curl::Easy.http_post(
  Settings::API_URL,
  fields.collect{|k, v| Curl::PostField.content(k.to_s, v)}
)
puts ch.body_str

File: html/redcap_v15.7.6/API/examples/perl/import_project.pl
#!/usr/bin/env perl

use strict;
use warnings;
use LWP::Curl;
use JSON;

my %config = do 'config.pl';

my $record = {
	project_title => 'Project ABC',
	purpose       => 0,
	purpose_other => '',
	project_notes => 'Some notes about the project'
};

my $json = JSON->new->allow_nonref;
my $data = $json->encode($record);

my $fields = {
    token   => $config{api_super_token},
    content => 'project',
    format  => 'json',
    data    => $data,
};

my $ch = LWP::Curl->new(auto_encode => 0);
my $content = $ch->post($config{api_url}, $fields, $config{referer});

print $content;
